### PR TITLE
Add merge support

### PR DESCRIPTION
**Pull request by qzchenwl**
I am using this in my SVG editor like project.

I moved shape A from position (x0, y0) to (x1, y1) , and then want to undo this operation. `ljur.undo` should change position to (x0, y0) instead of (x1 - 1, y1).

### CODE REVIEW

**Comment by azazdeaz** [comment]
Looks great! i'll need time to learn it before i merge.

Can you tell me more about your use case, and why doesn't the flagging enough to solve your problem?

**Comment by qzchenwl** [comment]
I am using [Polymer](https://www.polymer-project.org/0.5/) for data-binding. I don't explicitly change the data model, that is why I can't manually start/end flagging.

For example:
1). `<input type="text" value="{{shape.desc}}">`, the `shape.desc` is auto mutated when user change the text in the input element.

2). `<rect x="{{shape.x}}" y="{{shape.y}}"></rect>`. In this case, [ x mutated, y mutated, x mutated ... ]. So I need to regard shape.x and shaped.y as the same attribute, to group the series mutation.

**Comment by azazdeaz** [comment]
I see, thanks! Can you also clarify how do you using the ```mergeBlacklist```, ```mergeWhitelist```, ```mergeGlobalBlacklist```, and ```mergeGlobalWhitelist```?

File: LazyJsonUndoRedo.js [hunk 1]
```diff
@@ -426,10 +460,178 @@
...
+                this._mergeWhitelists.splice(i--, 1);
+            }
+        }
+    };
+
+    /**
+     * Set attributes whose change list will be merged
+     * @param obj - attributes in this object
+     * @param list - attribute list
+     */
```
Comment (qzchenwl) [inline LazyJsonUndoRedo.js]: Sorry for the wrong comment.

It should be `Set attributes whose change list will *not* be merged`

**Comment by qzchenwl** [comment]
`setMergeBlacklist(o, list)` means attributes in object `o` should be merged except for those in `list`.
`setMergeWhitelist(o, list)` means attriubtes in object `o` should *not* be merged(the default setting), except for those in `list`

`addToGlobalMergeBlacklist(attrs...)` means attributes in `attrs` should not be merged.
`addToGlobalMergeWhitelist(attrs...)` means attributes in `attrs` should be merged.

`setMergeGroups(o, [ [ 'x', 'y' ], [ 'w', 'h' ] ])` means `x` and `y` in `o` will be regard as the same attribute, and be merged together, the same for `w` and `h`.

`addToGlobalMergegroups([ 'x', 'y' ], [ 'w', 'h' ] )` means `x` and `y` in any same object will be regard as the same attribute, and be merged together, the same for `w` and `h`.

File: LazyJsonUndoRedo.js [hunk 2]
```diff
@@ -426,10 +460,178 @@
...
+    p.setMergeBlacklist = function (obj, list) {
```
Comment (qzchenwl) [inline LazyJsonUndoRedo.js]: Sorry for the wrong comment.

It should be Set attributes whose change list will *not* be merged

**Comment by azazdeaz** [comment]
Sorry for the delay, i merge it now!
I felt it a little bit hacky, mostly because if you have an object ```{a, b}``` and you merge ```a``` and ```b``` than the changes of that object will be considered as one step until you change a different observed value.
But i have to admit in real world use cases (ex yours as you described) it's more like the desired behavior than an issue. (And this lib is for hacking anyway :) )

Thank you for the cool new feature!

**Pull request merged**